include/spi.h

```diff
@@ -160,4 +191,8 @@ int32_t spi_write_and_read(struct spi_desc *desc,
 			   uint8_t *data,
 			   uint16_t bytes_number);
 
+/* Iterate over head list and send all spi messages. */
+int32_t spi_transfer(struct spi_desc *desc, struct spi_msg_list *head);
```

Comment:
First off, i think the user shouldn't need to care how you implemented this and whether they should pass the head or the tail or whatever as parameter. The naming of `head` then is rather unfortunate, better naming would be `msg_list` or `xfer_list`.